src/query/sql/src/planner/semantic/type_check.rs

```diff
@@ -1738,7 +1774,8 @@ impl<'a> TypeChecker<'a> {
             .collect::<Vec<_>>();
 
         // TODO: support multiple params
-        if params.len() != 1 {
+        // ARRAY_REDUCE have two params
+        if params.len() != 1 && func_name != "array_reduce" {
```

Comment:
We also need to check the length of params equal to 2, if the `func_name` is `array_reduce`